metau-capstone/Delegates/SceneDelegate.m

```diff
@@ -22,14 +25,30 @@ - (void)scene:(UIScene *)scene willConnectToSession:(UISceneSession *)session op
     PFUser *user = [PFUser currentUser];
     if (user != nil) {
         NSLog(@"Welcome back %@ 😀", user.username);
-
-        // Load Chat view controller and set as root view controller
         UIStoryboard *storyboard = [UIStoryboard storyboardWithName:@"Main" bundle:nil];
         UIViewController *tabBarNavigationController = [storyboard instantiateViewControllerWithIdentifier:@"tabBarController"];
         self.window.rootViewController = tabBarNavigationController;
     }
 }
 
+- (void)scene:(UIScene *)scene openURLContexts:(NSSet<UIOpenURLContext *> *)URLContexts {
+    NSURL *url = [URLContexts allObjects][0].URL;
+    NSArray *queryPair = [url.query componentsSeparatedByString:@"="];
+    NSString *userID = queryPair[1];
```

Comment:
In the future if you chose to add other url parameters (say custom decks, or a key to verify that you have permission to download someone's cards, etc) splitting the query string like this might always work.

Say we have `?key=123&userId=456` and `?userId=456&key=123`. We want to be able to deconstruct this and get the userId, but these hard rules (indexing first after the split) might not work consistently.

Reply:
Okay, I refactored the code to create a queryStringDictionary. Instead of getting the userID with a hard-coded index number, I get the value correlated with the @"userID" key.